--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -1772,3 +1772,5 @@
 	.sta_pre_rcu_remove = mt76_sta_pre_rcu_remove,
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 13, 0)
 	.sta_rc_update = mt7915_sta_rc_update,
+#endif
 	.set_key = mt7915_set_key,
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -1667,3 +1667,5 @@
 	.sta_pre_rcu_remove = mt76_sta_pre_rcu_remove,
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 13, 0)
 	.sta_rc_update = mt7996_sta_rc_update,
+#endif
 	.set_key = mt7996_set_key,
--- a/mac80211.c
+++ b/mac80211.c
@@ -613,7 +613,7 @@ int mt76_create_page_pool(struct mt76_dev *dev, struct mt76_queue *q)
 {
 	struct page_pool_params pp_params = {
 		.order = 0,
-		.flags = PP_FLAG_PAGE_FRAG,
+		.flags = 0,
 		.nid = NUMA_NO_NODE,
 		.dev = dev->dma_dev,
 	};