--- a/nf_nat_fullcone.c
+++ b/nf_nat_fullcone.c
@@ -1122,7 +1122,7 @@
 	enum ip_conntrack_info ctinfo;
 	struct nf_conntrack_tuple *ct_tuple_origin;
 
-	uint16_t port, original_port;
+	uint16_t port;
 	uint8_t protonum;
 
 /* NFPROTO specific def */
@@ -1143,7 +1143,6 @@
 	ip_6 = NULL;
 	/* NFPROTO specific init end */
 
-	original_port = 0;
 	ret = NFT_CONTINUE;	// BUG: use XT_CONTINUE for Xtables
 
 	ct = nf_ct_get(skb, &ctinfo);
@@ -1257,9 +1256,9 @@
 	enum ip_conntrack_info ctinfo;
 	struct nf_conn_nat *nat;
 	struct nf_conntrack_tuple *ct_tuple, *ct_tuple_origin;
-	uint16_t port, original_port, want_port;
+	uint16_t port, original_port, want_port = 0;
 	uint8_t protonum;
-	bool is_src_mapping_active;
+	bool is_src_mapping_active = false;
 
 	/* NFPROTO specific def */
 	struct nat_mapping *mapping, *src_mapping;
--- a/nft_ext_fullcone.c
+++ b/nft_ext_fullcone.c
@@ -121,7 +121,7 @@
 }
 #endif
 
-static int nft_fullcone_validate(const struct nft_ctx *ctx, const struct nft_expr *expr, const struct nft_data **data)
+static int nft_fullcone_validate(const struct nft_ctx *ctx, const struct nft_expr *expr)
 {
 	int err;
 
@@ -179,7 +179,7 @@
 	return err;
 }
 
-static int nft_fullcone_dump(struct sk_buff *skb, const struct nft_expr *expr)
+static int nft_fullcone_dump(struct sk_buff *skb, const struct nft_expr *expr, bool reset)
 {
 	const struct nft_fullcone *priv = nft_expr_priv(expr);
 
